client/package.json

```diff
@@ -1,61 +1,61 @@
-        "url": "^0.11.0"
-    },
-    "scripts": {
-        "start": "react-scripts start",
-        "build": "cross-os cross-os-build",
-        "eject": "react-scripts eject",
-        "pretest": "npm run build",
-        "preinstall": "npm i npm-force-resolutions --save-dev && npm-force-resolutions",
-        "test": "react-scripts test",
-        "posttest": "license-checker --summary",
-        "report-coverage": "nyc report --reporter=text-lcov > coverage.lcov"
-    },
-    "cross-os": {
-        "cross-os-build": {
-            "darwin": "npm i && react-scripts build && rm -rf ../server/client-build/ && mv build ../server/client-build/",
-            "linux": "npm i && react-scripts build && rm -rf ../server/client-build/ && mv build ../server/client-build/",
-            "win32": "npm i && react-scripts build && rm -rf ..\\server\\client-build && mv build ..\\server\\client-build"
-        }
-    },
-    "eslintConfig": {
-        "extends": "react-app"
-    },
-    "proxy": "http://localhost:9000",
-    "browserslist": {
-        "production": [
-            ">0.2%",
-            "not dead",
-            "not op_mini all"
-        ],
-        "development": [
-            "last 1 chrome version",
-            "last 1 firefox version",
-            "last 1 safari version"
-        ]
-    },
-    "devDependencies": {
-        "codecov": "^3.7.0",
-        "license-checker": "^25.0.1",
-        "npm-force-resolutions": "0.0.3"
-    },
-    "resolutions": {
-        "@react/collection-view": "file:libraries/react-collection-view-4.1.5.tgz"
-    }
+	"name": "client",
+	"version": "0.1.0",
+	"private": true,
+	"dependencies": {
+		"@react/collection-view": "file:libraries/react-collection-view-4.1.5.tgz",
+		"@react/react-spectrum": "file:libraries/react-react-spectrum-2.25.0.tgz",
+		"ace-builds": "^1.4.8",
+		"ajv": "^6.12.0",
+		"babel-polyfill": "^6.26.0",
+		"cross-os": "^1.3.0",
+		"form-data": "^3.0.0",
+		"js-yaml": "^3.13.1",
+		"react": "^16.13.0",
+		"react-ace": "^8.0.0",
+		"react-dom": "^16.13.0",
+		"react-scripts": "3.4.0",
+		"url": "^0.11.0"
+	},
+	"scripts": {
+		"start": "PORT=3000 react-scripts start",
+		"build": "cross-os cross-os-build",
+		"eject": "react-scripts eject",
+		"pretest": "npm run build",
+		"preinstall": "npm i npm-force-resolutions --save-dev && npm-force-resolutions",
+		"test": "react-scripts test",
+		"posttest": "license-checker --summary",
+		"report-coverage": "nyc report --reporter=text-lcov > coverage.lcov"
+	},
+	"cross-os": {
+		"cross-os-build": {
+			"darwin": "npm i && react-scripts build && rm -rf ../server/client-build/ && mv build ../server/client-build/",
+			"linux": "npm i && react-scripts build && rm -rf ../server/client-build/ && mv build ../server/client-build/",
+			"win32": "npm i && react-scripts build && rm -rf ..\\server\\client-build && mv build ..\\server\\client-build"
+		}
+	},
+	"eslintConfig": {
+		"extends": "react-app"
+	},
+	"proxy": "http://localhost:3000",
```

Comment:
the proxy port (backend port) has to be different than the port the frontend is running on. The backend + frontend cannot run on the same port.

see article here explaining how to proxy backend requests in react apps: https://www.twilio.com/blog/react-app-with-node-js-server-proxy

Reply:
The front end and back end don't run on the same port when in development mode, and that's what seemed to read in the original code. The only time the proxy was changed was when set-port.js was called and that was only called when run client was called and that was only called in development mode. 

So are you saying that in both dev and prod mode, we should be writing the proxy port?

Reply:
we should NOT be writing to the package.json in prod mode. To clarify, the proxy is ONLY used in development mode.  The point of developer mode is to start up 2 local servers: 1 for the client and 1 for the server. By setting the `proxy` in the package.json, you proxy all api requests (backend requests) to the port in `proxy`. Therefore, the port listed in `proxy` is the backend port used in the development mode. The only way to change the proxy port is by changing the package.json. This is why I made the `set-port.js` script. It changes the proxy port in the client package.json and starts up the backend server with the specified port.

you added both:
```
"start": "PORT=3000 react-scripts start",
```
and 
```
"proxy": "http://localhost:3000",
```
which would make the client run on port 3000 and proxy backend requests also to port 3000.

Reply:
Ok I misunderstood your original comment that those values should not be default in package.json, I will remove those.

We are not writing to package.json in prod mode at all.
I do understand about the differences between prod and dev for server and client.
Both the server and client ports are coded to be different when running in dev mode.

Did I misunderstand anything else?

Reply:
Yes, that is correct. Sorry for the misunderstanding. It is sometimes hard to express things clearly online.